cache/blobs.go

```diff
@@ -184,6 +172,19 @@ func computeBlobChain(ctx context.Context, sr *immutableRef, createIfNeeded bool
 						enableOverlay = false
 					}
 				}
+				mediaType := comp.Type.MediaType()
+				if sr.cm.Snapshotter.Name() == "overlaybd" {
+					snStat, err := sr.cm.Snapshotter.Stat(ctx, sr.getSnapshotID())
+					if err != nil {
+						return nil, errors.Wrapf(err, "failed to Stat overlaybd")
+					}
+					if bdPath := snStat.Labels[obdlabel.LocalOverlayBDPath]; bdPath != "" {
+						if err := commitOverlayBD(ctx, sr, &desc); err != nil {
+							return nil, err
+						}
+						mediaType = desc.MediaType
+					}
+				}
```

Comment:
I tried to move this closer to where it's used, but wasn't sure if `enableOverlay` (below) is _possible_ with this snapshotter. It's disabled by default for `overlaybd`, but can be enabled with the `BUILDKIT_DEBUG_FORCE_OVERLAY_DIFF` env-var.

If it is not supported, then probably there should be an error, and otherwise we may have to split this block (for the mediaType part), and move the `commit` back to below.

Reply:
Acturally, `commitOverlayBD ` do similar things like `tryComputeOverlayBlob` but is for block device. If `enableOverlay ` is true, `commitOverlayBD` should overcover desc generated by `tryComputeOverlayBlob`. We set `enableOverlay` to false to avoid this (because it will make overlaybd build slower).

Reply:
@HileQAQ Thanks, yes that was the part I was wondering about; should we ignore `BUILDKIT_DEBUG_FORCE_OVERLAY_DIFF` for `overlaybd` altogether? We disable `enableOverlay` by default, but (technically) the `BUILDKIT_DEBUG_FORCE_OVERLAY_DIFF` could still enable this (and perhaps that should either produce an error, or a warning and be ignored?)

Reply:
Enabling `enableOverlay` has no effect on the result of the overlaybd build, it just makes it a little slower(in compute diff between lower and upper). So I think it shouldn't be an error, even don't need a warning.

Reply:
Gotcha; so, yes, perhaps we should disable it altogether, or put it in a branch where it's `!= "overlaybd"`

Reply:
Yes, I will merge this PR and make some changes related to this.